src/components/countryDetails/CountryDetails.tsx

```diff
@@ -1,45 +1,214 @@
-import {
-	Link,
-	LoaderFunction,
-	useLoaderData,
-	useParams,
-} from "react-router-dom";
+import { useEffect } from "react";
+import { LoaderFunction, useLoaderData } from "react-router-dom";
 import { useCountriesContext } from "../../Context/CountriesDataContext";
+import {
+	StyledArrowIcon,
+	StyledCountryName,
+	StyledDivWithButtons,
+	StyledDivWithCountryData,
+	StyledDivWithCountryInfo,
+	StyledDivWithFlag,
+	StyledFlagImage,
+	StyledLeftColumnWithCountryInfo,
+	StyledLink,
+	StyledLinkWithBorderCountries,
+	StyledParagraphWithCountryInfo,
+	StyledSection,
+	StyledSpanWithCategoryName,
+} from "./CountryDetails.styles";
+import { CountriesProps } from "../CountriesList/CountriesProps";
 
 export interface CountryDataTypes {
 	name: {
 		common: string;
+		nativeName: {
+			ara: {
+				official: string;
+			};
+		};
+	};
+	altSpellings: string;
+	population: number;
+	flags: {
+		alt: string;
+		png: string;
+	};
+	region: string;
+	subregion: string;
+	capital: [];
+	tld: string[];
+	currencies: {
+		[key: number]: {
+			name: string;
+		};
 	};
+
+	languages: string;
+	borders: string[];
 }
 
 export const CountryDetails = () => {
-	const { inputValue, setInputValue } = useCountriesContext();
-	const { commonName } = useParams();
+	const { inputValue, setInputValue, countries, fetchCountries } =
+		useCountriesContext();
 	const countryData = useLoaderData() as CountryDataTypes[];
 
+	useEffect(() => {
+		if (!countries?.length) {
+			fetchCountries();
+		}
+	}, []);
+
 	const handleClick = (): void => {
 		setInputValue(inputValue);
 	};
 
 	return (
-		<div>
-			<Link onClick={handleClick} to="/">
-				Go back
-			</Link>
-			countryDetails
-			<p>{commonName}</p>
-			<div>
-				{countryData?.map((country) => {
-					return <p key={country.name.common}>{country.name.common}</p>;
-				})}
-			</div>
-		</div>
+		<StyledSection>
+			<StyledDivWithFlag>
+				<StyledLink onClick={handleClick} to="/">
+					<StyledArrowIcon />
+					Back
+				</StyledLink>
+
+				<div>
+					{countryData?.map(({ name, flags }) => {
+						return (
+							<StyledFlagImage
+								src={flags.png}
+								alt={flags.alt}
+								key={name.common}
+							/>
+						);
+					})}
+				</div>
+			</StyledDivWithFlag>
+			<StyledDivWithCountryData>
+				{countryData?.map(
+					({
+						name,
+						population,
+						region,
+						subregion,
+						capital,
+						tld,
+						currencies,
+						languages,
+						altSpellings,
+						borders,
+					}) => {
+						const arr: CountriesProps[] = [];
+
+						borders?.forEach((singleBorder) => {
+							const borderCountry = countries?.find((singleCountry) => {
+								return singleCountry.cca3 === singleBorder;
+							}) as CountriesProps;
+
+							arr.push(borderCountry);
+						});
+
+						const currencieKey = Object.keys(currencies)[0];
+
+						const currencieName =
+							currencies[currencieKey as unknown as number].name;
```

Comment:
Please try to get rid of as unknows as x. This looks like a hack that we intruduced just to make TS happy

Reply:
fixed.